src/app/project/project.service.ts

```diff
@@ -1,142 +1,223 @@
-      project_id: 55,
-      progress :1,
-      description: ' You are now part of the learning community. Whether you are a Project Stakeholder, Consulting Mentor or a University Student about to embark on your first consulting project, welcome. This Fundamentals milestone will provide you with an overview, explain your role and provide you with tips and tricks for a successful learning experience ',
-      Activity :[
-        {
-          id: 101,
-          name: 'Test Activity one',
-          isLocked: false,
-          progress: 0.34,
-          hasFeedback: true,
-          leadImage:''
-        },
-        {
-          id: 102,
-          name: 'Activity two',
-          isLocked: true,
-          progress: 0.74,
-          leadImage: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSGTVf63Vm3XgOncMVSOy0-jSxdMT8KVJIc8WiWaevuWiPGe0Pm',
-          hasFeedback: false,
-        },
-        {
-          id: 102,
-          name: 'Activity two',
-          isLocked: true,
-          progress: 0.74,
-          leadImage: '',
-          hasFeedback: false,
-        } 
-      ],
-      isLocked: false
-    },
-    {
-      id: 2,
-      name: 'delivery',
-      description:'', 
-      project_id: 55,
-      progress: 0.62,
-      Activity :[
-        { 
-          id: 103,
-          name: 'Test Activity three',
-          isLocked: false,
-          progress: 0.47,
-          hasFeedback: false,
-        },
-        {
-          id: 104,
-          name: 'Activity four',
-          isLocked: false,
-          progress:0.98,
-          hasFeedback: false,
-        },
-        {
-          id: 105,
-          name: 'Test Activity five',
-          isLocked: false,
-          progress: 0.47,
-          hasFeedback: false,
-        },
-      ],
-      isLocked: false
-    },
-    {
-      id: 3,
-      name: 'communication',
-      description: 'there is some dummy text',
-      progress: 1,
-      project_id: 55,
-      Activity :[
-        {
-          id: 106,
-          name: 'Test Activity six',
-          isLocked: false,
-          progress: 1,
-          hasFeedback: false,
-        },
-        {
-          id: 107,
-          name: 'Activity seven',
-          isLocked: false,
-          progress:0.48,
-          hasFeedback: true,
+  public milestones: Array<Milestone> = [];
+  public milestone_ids: Array<number> = [];
+  public activities: Array<Activity> = [];
+  
+  constructor( 
+    private request: RequestService,
+    private utils: UtilsService) { }
+  
+  public getMilestones() {
+    return this.request.get(api.milestone)
+    .pipe(map(response => {
+      if (response.success && response.data) {
+        return this._normaliseMilestones(response.data);
+      }
+    }));
+  }
+  
+  private _normaliseMilestones(data){
+    
+    if (!Array.isArray(data)) {
+      this.request.apiResponseFormatError('Milestones array format error');
+      return [];
+    }
+    this.milestones = [];
+    this.milestone_ids = [];
```

Comment:
Don't need to define `milestones` and `milestone_ids` as global variables since we just use those within this function.
Note:
Do not define global variables if we don't have to. This will cause the problem below:
1. Use more memory to store the global variable
2. The problem that you've face before. Because these are global variables, after switch to activity page, `milestones` still exist with the previous value. If you didn't assign empty array here, milestones will be added upon the previous one. If we didn't define these as global variable, those variable won't exist anymore when you come back from activity page, so that this problem won't happen
3. Also, it is not clear that where have we touched this global variable